File: sonar-dotnet-core/src/test/resources/samples/vbnet/Calculator/Calculator.NUnit3/CalculatorTests.vb
Imports NUnit.Framework

<TestFixture>
Public Class Tests
    <Test>
    Public Sub TestMethod1()
        Dim calculator As New Calculator()
        Dim result = calculator.Add(1, 2, Function(x) x > 0)
        Assert.That(result, [Is].EqualTo(3))
    End Sub
End Class

<TestFixture>
Public Class BaseClass(Of T As Class)
    <Test>
    Public Sub TestMethodInBaseClass()
        Assert.AreEqual(1, 1)
    End Sub

    <Test>
    Public Overridable Sub VirtualMethodInBaseClass()
        Assert.AreEqual(1, 1)
    End Sub
End Class

<TestFixture>
Public NotInheritable Class Derived
    Inherits BaseClass(Of String)

    <Test>
    Public Overrides Sub VirtualMethodInBaseClass()
        Assert.AreEqual(1, 1)
    End Sub
End Class

<TestFixture>
Public NotInheritable Class GenericCalculatorTests(Of T As Class)
    Inherits BaseClass(Of T)

    <Test>
    Public Sub Method()
        Assert.AreEqual(1, 1)
    End Sub

    <Test>
    Public Sub GenericMethod(Of T)()
        Assert.AreEqual(1, 1)
    End Sub

    <Test>
    Public Overrides Sub VirtualMethodInBaseClass()
        Assert.AreEqual(1, 1)
    End Sub
End Class

<TestFixture>
Public Class GenericTests
    <TestCase(42)>
    <TestCase("string")>
    <TestCase(Double.Epsilon)>
    Public Sub GenericTest(Of T)(instance As T)
        Console.WriteLine(instance)
    End Sub
End Class

File: sonar-dotnet-core/src/test/resources/samples/csharp/Calculator/Calculator.NUnit3/CalculatorTests.cs
namespace Calculator.NUnit3;

[TestFixture]
public class Tests
{
    [Test]
    public void TestMethod1()
    {
        var calculator = new Calculator();
        var result = calculator.Add(1, 2, x => x > 0);
        Assert.That(result, Is.EqualTo(3));
    }
}

[TestFixture]
public class BaseClass<T> where T : class
{
    [Test]
    public void TestMethodInBaseClass() =>
        Assert.AreEqual(1, 1);

    [Test]
    public virtual void VirtualMethodInBaseClass() =>
        Assert.AreEqual(1, 1);
}

[TestFixture]
public sealed class Derived : BaseClass<string>
{
    [Test]
    public override void VirtualMethodInBaseClass() =>
        Assert.AreEqual(1, 1);
}

[TestFixture]
public sealed class GenericCalculatorTests<T> : BaseClass<T> where T : class
{
    [Test]
    public void Method() =>
        Assert.AreEqual(1, 1);

    [Test]
    public void GenericMethod<T>() =>
        Assert.AreEqual(1, 1);

    [Test]
    public override void VirtualMethodInBaseClass() =>
        Assert.AreEqual(1, 1);
}

[TestFixture]
public class GenericTests
{
    [TestCase(42)]
    [TestCase("string")]
    [TestCase(double.Epsilon)]
    public void GenericTest<T>(T instance)
    {
        Console.WriteLine(instance);
    }
}
